kornia/feature/adalam/adalam.py

```diff
@@ -168,18 +171,18 @@ def filter_matches(
 
     def match_and_filter(
         self,
-        k1,
-        k2,
-        d1,
-        d2,
-        im1shape=None,
-        im2shape=None,
-        o1=None,
-        o2=None,
-        s1=None,
-        s2=None,
+        k1: Union[Tensor, 'npt.NDArray[Any]'],
```

Comment:
Not sure if we want to support arrays here

Reply:
Ideally not yet numpy support

Reply:
Originally was being supported arrays here, I think it should be supported only Tensors too, but I wanted to avoid breaking things in this PR -- but if you don't mind I can drop the support here

Had a method `__to_tensor` which I updated to use `as_tensor` to keep the support